staging/src/k8s.io/client-go/tools/cache/shared_informer.go

```diff
@@ -149,7 +153,20 @@ type SharedInformer interface {
 	// between any two resyncs may be longer than the nominal period
 	// because the implementation takes time to do work and there may
 	// be competing load and scheduling noise.
-	AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration)
+	// It returns a registration handle for the handler that can be used to remove
+	// the handler again and an error if the handler cannot be added.
+	AddEventHandlerWithResyncPeriod(handler ResourceEventHandler, resyncPeriod time.Duration) (ResourceEventHandlerRegistration, error)
+	// RemoveEventHandlerByRegistration removes a formerly added event handler given by
+	// its registration handle.
+	// If, for some reason, the same handler has been added multiple
+	// times, only the registration for the given registration handle
+	// will be removed.
+	// It returns an error if the handler cannot be removed,
+	// because the informer is already stopped.
+	// Calling Remove on an already removed handle returns no error
+	// because the handler is finally (still) removed after calling this
+	// method.
+	RemoveEventHandlerByRegistration(handle ResourceEventHandlerRegistration) error
```

Comment:
I'd leave off the "ByRegistration" part -- you can tell what you need by the function signature.

Reply:
suffix removed